cal/cal/static/js/components/category-details.js

```diff
@@ -63,61 +61,34 @@ function CategoriesDetailCtrl($scope, $http, QueryService){
   }.bind(this);
 
   this.showDaily = function() {
-    $http({
-      method: 'GET',
-      url: this.timeseriesDay + '.json',
-      params: {
-        timezone: query_timezone,
-      }
-    }).
-    then(function successCallback(response) {
+    QueryService.populateDay('Category day' + _this.categoryId, 'Category', _this.categoryId, []).
+    then(function populate(data) {
       _this.timeStep = "day";
-      _this.averageHours = Math.round(((_this.categoryHours / response.data[0].length) * 100)) / 100;
-      var eventData = QueryService.populateData(response.data, 'Category');
-      _this.ctrlGraphData = eventData[0];
-      _this.showGraph(eventData[1]);
+      _this.averageHours = Math.round(((_this.categoryHours / data[0][0].values.length) * 100)) / 100;
+      _this.ctrlGraphData = data[0];
+      _this.showGraph(data[1]);
+      _this.dailyData = data;
     }, function errorCallback() {
       throw "Failed to get timeseries day";
     });
   };
 
   this.showWeekly = function() {
-    $http({
-      method: 'GET',
-      url: this.timeseriesWeek + '.json',
-      params: {
-        timezone: query_timezone,
-      }
-    }).
-    then(function successCallback(response) {
-      _this.timeStep = "week";
-      // round(... * 100) / 100 neceessary to round average hours to two decimal
-      _this.averageHours = Math.round(((_this.categoryHours / response.data[0].length) * 100)) / 100;
-      var eventData = QueryService.populateData(response.data, 'Category');
-      _this.ctrlGraphData = eventData[0];
-      _this.showGraph(eventData[1]);
-    }, function errorCallback() {
-      throw "Failed to get timeseries week";
-    });
+    var data = QueryService.populateData('Category week' + _this.categoryId, 'Category', _this.categoryId, "week", _this.dailyData);
+    _this.timeStep = "week";
+    var numWeeks = data[0][0].values.length;
```

Comment:
Same comment as above, except this time with the double array. Try to make everything nested dicts, because anyone coming after you will have no idea what is going on from reading this code alone.